clients/src/test/java/org/apache/kafka/common/telemetry/internals/MetricNamingConventionTest.java

```diff
@@ -0,0 +1,196 @@
+            Collections.emptyMap());
+        MetricKey metricKey = metricNamingStrategy.metricKey(metricName);
+
+        // '-metrics' gets removed from the group name.
+        assertEquals("org.apache.kafka.group.name.metrics", metricKey.getName());
+        assertEquals(Collections.emptyMap(), metricKey.tags());
+    }
+
+    @Test
+    public void testMetricKeyWithNullTagKey() {
+        MetricName metricName = new MetricName("name", "group", "description",
+            Collections.singletonMap(null, "value1"));
+        Exception e = assertThrows(NullPointerException.class, () -> metricNamingStrategy.metricKey(metricName));
+        assertEquals("metric data cannot be null", e.getMessage());
+    }
+
+    @Test
+    public void testMetricKeyWithBlankTagKey() {
+        MetricName metricName = new MetricName("name", "group", "description",
+            Collections.singletonMap("", "value1"));
+        MetricKey metricKey = metricNamingStrategy.metricKey(metricName);
+
+        assertEquals("org.apache.kafka.group.name", metricKey.getName());
+        assertEquals(Collections.singletonMap("", "value1"), metricKey.tags());
+    }
+
+    @Test
+    public void testDerivedMetricKey() {
+        MetricName metricName = new MetricName("name", "group", "description",
+            Collections.emptyMap());
+        MetricKey metricKey = metricNamingStrategy.derivedMetricKey(
+            metricNamingStrategy.metricKey(metricName), "delta");
+
+        assertEquals("org.apache.kafka.group.name.delta", metricKey.getName());
+        assertEquals(Collections.emptyMap(), metricKey.tags());
+    }
+
+    @Test
+    public void testDerivedMetricKeyWithTags() {
+        MetricName metricName = new MetricName("name", "group", "description",
+            Collections.singletonMap("tag1", "value1"));
+        MetricKey metricKey = metricNamingStrategy.derivedMetricKey(
+            metricNamingStrategy.metricKey(metricName), "delta");
+
+        assertEquals("org.apache.kafka.group.name.delta", metricKey.getName());
+        assertEquals(Collections.singletonMap("tag1", "value1"), metricKey.tags());
+    }
+
+    @Test
+    public void testDerivedMetricKeyWithNullComponent() {
+        MetricName metricName = new MetricName("name", "group", "description",
+            Collections.emptyMap());
+        Exception e = assertThrows(NullPointerException.class, () -> metricNamingStrategy.derivedMetricKey(
+            metricNamingStrategy.metricKey(metricName), null));
+        assertEquals("derived component cannot be null", e.getMessage());
+    }
+
+    @Test
+    public void testDerivedMetricKeyWithBlankComponent() {
+        MetricName metricName = new MetricName("name", "group", "description",
+            Collections.emptyMap());
+        MetricKey metricKey = metricNamingStrategy.derivedMetricKey(
+            metricNamingStrategy.metricKey(metricName), "");
+
+        // Ends with dot, though derived component should
```

Comment:
Looks like half a sentence in this comment.

Reply:
Completed it, sorry.